pkg/runtime/wfengine/workflow.go

```diff
@@ -387,11 +391,13 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	case <-ctx.Done(): // caller is responsible for timeout management
 		return ctx.Err()
 	case completed := <-callback:
+		// [Question] Is this rescheduling the workflow to be completed ?
```

Comment:
If the workflow execution has completed successfully, then `true` is returned on the channel. In case of any errors, `false` is returned, and then recoverable error returned makes the engine retry it again(https://github.com/dapr/dapr/blob/HEAD/pkg/runtime/wfengine/workflow.go#L118-L119)